app/src/main/java/org/catroid/catrobat/newui/ui/adapter/LookAdapter.java

```diff
@@ -8,19 +12,46 @@
 
 public class LookAdapter extends RecyclerViewAdapter<LookInfo> {
 
+
     public LookAdapter(List<LookInfo> lookInfos, int listItem) {
         super(lookInfos, listItem);
     }
 
     @Override
-    public void bindDataToViewHolder(LookInfo item, ViewHolder holder, boolean isSelected) {
+    public void bindDataToViewHolder(LookInfo item, ViewHolder holder, final boolean isSelected) {
         holder.mNameView.setText(item.getName());
         holder.mDetailsView.setText("");
+        final ViewHolder holderCopy = holder;
+        final LookInfo itemCopy = item;
+        //final Context view_context = holder.mItemView.getContext();
+
+
+        in.setAnimationListener(new Animation.AnimationListener(){
+            @Override
+            public void onAnimationStart(Animation animation) {
+            }
+            @Override
+            public void onAnimationEnd(Animation animation) {
+                if(isSelected) {
+                    holderCopy.mImageSwitcher.setImageResource(R.drawable.ic_check_circle_black_24dp);
+
+                } else {
+                    Bitmap croppedBitmap = itemCopy.getCroppedBitmap(itemCopy.getBitmap());
```

Comment:
You should use the LookInfo's getThumbnail() instead. You should integrate the cropping into this method (possibly by just calling it from there).